ui/src/app/shared/services/requests.ts

```diff
@@ -36,7 +37,15 @@ export default {
         apiRoot = val;
     },
     agent,
+    csrfHeaderName,
     onError: onError.asObservable().filter(err => err != null),
+
+    // Get CSRF Token by performing GET request. Returns token as string.
+    async getCsrfToken(): Promise<string> {
+        const resp = await initHandlers(agent.get(`${apiRoot}/session/userinfo`));
```

Comment:
In what sense? The UI client (browser) needs to get a CSRF token, which is supplied in the HTTP headers (`X-CSRF-Token`) of a response to an _authenticated_ GET request. We can turn it into a HEAD request, which indeed could be better the better verb. However, since the user is already authenticated, they only retrieve their own data. Or am I missing something?